include/lo2s/trace/trace.hpp

```diff
@@ -270,6 +270,11 @@ class Trace
                     ByProcess(groups_.get_process(scope.as_thread())))
                 .add_member(intern_location);
         }
+
+        // You can not have a location withouth a writer, so create one here. calling archive_ on
+        // the same location twice is harmless in case some requires a real writer later on
+        archive_(intern_location);
+
```

Comment:
The code is unnecessary confusing here. If metrics are used, a metric instance is created with the recorder location set to the location of the sample_writer (which is what location() creates). With `--system-wide` it can now happen that no sample_writer (and associated otf2-writer) will ever be created as with `--system-wide` it is now possible to disable everything that the sample_writer records, leaving us with a location (created for the metric_instance) which has no otf2-writer, which is something OTF2 really does not like. I alleviate this problem by always creating an otf2-writer for the sample_writer location, even if we dont plan to write something to it.

That being said, the way the location for the sample writer is created is unnecessary complex and I've changed this.